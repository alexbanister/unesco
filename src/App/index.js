import React, { Component } from 'react';
import { Route } from 'react-router-dom';
import Welcome from '../Welcome/';
import Dashboard from '../Dashboard/';
import SiteList from '../SiteList/';
import Profile from '../Profile/';

class App extends Component {
  render() {
    return (
      <div className="app">
        <main>
          <Route
            exact path='/'
            component={Welcome}
            key='Welcome'/>
          <Route
            exact path='/dashboard'
            component={Dashboard}
            key='Dashboard' />
          <Route
            exact path='/favorites'
            render={() => <SiteList title='Favorites' section='favorties'/>}
            key='Favorites' />
          <Route
            exact path='/visited'
            render={() => <SiteList title='Visited' section='visited' />}
            key='Visited' />
          <Route
            exact path='/wants'
            render={() => <SiteList title='Want To Go' section='wants' />}
            key='Wants' />
          <Route
            exact path='/profile'
            component={Profile}
            key='Profile' />
        </main>
      </div>
    );
  }
}

export default App;
