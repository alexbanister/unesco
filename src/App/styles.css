a {
  color: #5b92e5;
  text-decoration: none; }
  a:hover {
    color: #C08497; }

.tooltip {
  position: relative;
  display: inline-block; }

.tooltip .tooltiptext {
  visibility: hidden;
  width: 120px;
  background-color: black;
  color: #fff;
  text-align: center;
  padding: 5px 0;
  border-radius: 6px;
  top: 100%;
  left: 50%;
  margin-left: -60px;
  position: absolute;
  z-index: 1; }

.tooltip:hover .tooltiptext {
  visibility: visible; }

.tooltip .tooltiptext::after {
  content: " ";
  position: absolute;
  bottom: 100%;
  /* At the top of the tooltip */
  left: 50%;
  margin-left: -5px;
  border-width: 5px;
  border-style: solid;
  border-color: transparent transparent black transparent; }
