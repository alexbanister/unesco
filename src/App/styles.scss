@import "../variables.scss";

a {
  color: $highlight;
  text-decoration: none;
  &:hover {
    color: $highlight-secondary;
  }
}
.tooltip {
    position: relative;
    display: inline-block;
}

.tooltip .tooltiptext {
    visibility: hidden;
    width: 10rem;
    background-color: $dark;
    color: $background;
    text-align: center;
    padding: .5rem;
    border-radius: .25rem;
    top: 100%;
    left: 50%;
    margin-left: -5.5rem;
    position: absolute;
    z-index: 1;
}

.tooltip:hover .tooltiptext {
    visibility: visible;
}

.tooltip .tooltiptext::after {
    content: " ";
    position: absolute;
    bottom: 100%;  /* At the top of the tooltip */
    left: 50%;
    margin-left: -5px;
    border-width: 5px;
    border-style: solid;
    border-color: transparent transparent black transparent;
}
