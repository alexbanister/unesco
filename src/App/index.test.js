import React from 'react';
import { shallow } from 'enzyme';
import App from './index';


describe('App snapshot', () => {
  it('should always match the snapshot', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
