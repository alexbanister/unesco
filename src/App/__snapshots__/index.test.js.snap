// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should always match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <main>
        <Switch>
                <Route
                        component={[Function]}
                        exact={true}
                        path="/"
                />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/dashboard"
                />
                <Route
                        exact={true}
                        path="/favorites"
                        render={[Function]}
                />
                <Route
                        exact={true}
                        path="/visited"
                        render={[Function]}
                />
                <Route
                        exact={true}
                        path="/wants"
                        render={[Function]}
                />
                <Route
                        exact={true}
                        path="/explore"
                        render={[Function]}
                />
                <Route
                        component={[Function]}
                        exact={true}
                        path="/site/:id"
                />
                <Route
                        component={[Function]}
                />
        </Switch>
</main>,
      "className": "app",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Switch>
          <Route
                    component={[Function]}
                    exact={true}
                    path="/"
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/dashboard"
          />
          <Route
                    exact={true}
                    path="/favorites"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/visited"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/wants"
                    render={[Function]}
          />
          <Route
                    exact={true}
                    path="/explore"
                    render={[Function]}
          />
          <Route
                    component={[Function]}
                    exact={true}
                    path="/site/:id"
          />
          <Route
                    component={[Function]}
          />
</Switch>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/dashboard"
/>,
            <Route
              exact={true}
              path="/favorites"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/visited"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/wants"
              render={[Function]}
/>,
            <Route
              exact={true}
              path="/explore"
              render={[Function]}
/>,
            <Route
              component={[Function]}
              exact={true}
              path="/site/:id"
/>,
            <Route
              component={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Welcome",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Dashboard",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Favorites",
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/favorites",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Visited",
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/visited",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Wants",
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/wants",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Search",
            "nodeType": "class",
            "props": Object {
              "exact": true,
              "path": "/explore",
              "render": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "SiteDetails",
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/site/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "main",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <main>
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/dashboard"
                    />
                    <Route
                              exact={true}
                              path="/favorites"
                              render={[Function]}
                    />
                    <Route
                              exact={true}
                              path="/visited"
                              render={[Function]}
                    />
                    <Route
                              exact={true}
                              path="/wants"
                              render={[Function]}
                    />
                    <Route
                              exact={true}
                              path="/explore"
                              render={[Function]}
                    />
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/site/:id"
                    />
                    <Route
                              component={[Function]}
                    />
          </Switch>
</main>,
        "className": "app",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Switch>
            <Route
                        component={[Function]}
                        exact={true}
                        path="/"
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/dashboard"
            />
            <Route
                        exact={true}
                        path="/favorites"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/visited"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/wants"
                        render={[Function]}
            />
            <Route
                        exact={true}
                        path="/explore"
                        render={[Function]}
            />
            <Route
                        component={[Function]}
                        exact={true}
                        path="/site/:id"
            />
            <Route
                        component={[Function]}
            />
</Switch>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/dashboard"
/>,
              <Route
                exact={true}
                path="/favorites"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/visited"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/wants"
                render={[Function]}
/>,
              <Route
                exact={true}
                path="/explore"
                render={[Function]}
/>,
              <Route
                component={[Function]}
                exact={true}
                path="/site/:id"
/>,
              <Route
                component={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Welcome",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Dashboard",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/dashboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Favorites",
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/favorites",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Visited",
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/visited",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Wants",
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/wants",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Search",
              "nodeType": "class",
              "props": Object {
                "exact": true,
                "path": "/explore",
                "render": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "SiteDetails",
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/site/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "main",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
